### PR TITLE
Service page fix

### PR DESCRIPTION
**Pull request by AngryBerryMS**
Discussed with Tianqi, when the "LandingPageType" is `Root`, the generated service page's name should be "index.yml"，and we shouldn't create a folder/directory, either. (Only create folder/directory when the "LandingPageType" is `Service`)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/6484)

### CODE REVIEW

File: src/docfx/build/toc/ServicePageGenerator.cs [hunk 1]
```diff
@@ -86,15 +96,21 @@ internal class ServicePageGenerator
...
             }
 
             foreach (var item in node.Items)
             {
-                GenerateServicePageFromTopLevelTOC(item, results, $"{directoryName}/{filename}");
+                if (!string.IsNullOrEmpty(node.LandingPageType.ToString()) && node.LandingPageType == LandingPageType.Root)
+                {
+                    GenerateServicePageFromTopLevelTOC(item, results, $"{directoryName}");
+                }
+                else
```
Comment (live1206) [inline src/docfx/build/toc/ServicePageGenerator.cs:110]: if `node.LandingPageType` is empty, shall we generate service page for it?

Comment (AngryBerryMS) [reply src/docfx/build/toc/ServicePageGenerator.cs:110]: Although `node.LandingPageType` may be empty, it's sub-sub items may have `LandingPageType`.
Although I haven't found cases like this, I think it deserves to be considered.........

Comment (live1206) [reply src/docfx/build/toc/ServicePageGenerator.cs:110]: Then we can just remove `!string.IsNullOrEmpty(node.LandingPageType.ToString())` here

Comment (AngryBerryMS) [reply src/docfx/build/toc/ServicePageGenerator.cs:110]: image

Comment (AngryBerryMS) [reply src/docfx/build/toc/ServicePageGenerator.cs:110]: Fixed.

**Review by live1206 — APPROVED** [review]

File: .github/actions/spell-check/dictionary/dictionary.txt [hunk 2]
```diff
@@ -189829,6 +189829,7 @@ ileosigmoidostomy
 ileostomies
 ileostomy
 ileotomy
+Ilearn
```
Comment (yufeih) [inline .github/actions/spell-check/dictionary/dictionary.txt:189832]: revert

**Review by yufeih — APPROVED** [review]

**Pull request merged**